### PR TITLE
Improvment particle system

### PR DESCRIPTION
**Pull request by IbraheemOsama**
Covering the Sphere and cone from issue #2971
[cone.txt](https://github.com/BabylonJS/Babylon.js/files/1590071/cone.txt)
[sphere.txt](https://github.com/BabylonJS/Babylon.js/files/1590072/sphere.txt)

### CODE REVIEW

File: src/Particles/babylon.particleSystem.ts [hunk 1]
```diff
@@ -137,6 +140,106 @@
         }
         // end of sheet animation
 
+        // Emitter Type
+
+        /**
+         * This bool is used to specify whether to use the default sphere calculated direction or to calculate the direction using direction1 and direction2 properties.
```
Comment (deltakosh) [inline src/Particles/babylon.particleSystem.ts]: Hey it's me the picky guy :D
As we will probably add more in the future I would recommend to provide an interface like IParticleEmitterType which provide a startPositionFunction

And then we can provide 3 emitters (box, sphere and cone)

Also we can provide static on the ParticleSystem class like CreateBoxEmitter(size) or CreateSphereEmitter(radius)

**Comment by IbraheemOsama** [comment]
But this is going to be a breaking change then 😁  are you ok with that ?

Thanks,
Ibraheem Osama
________________________________

**Comment by deltakosh** [comment]
Not necessary as the box mode can just rely on current direction1/2 properties

File: src/Particles/babylon.boxParticleEmitter.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+module BABYLON {
+    export interface IParticleEmitterType {
```
Comment (deltakosh) [inline src/Particles/babylon.boxParticleEmitter.ts]: Could you move this to a separate file?

File: src/Particles/babylon.boxParticleEmitter.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+module BABYLON {
+    export interface IParticleEmitterType {
+        startDirectionFunction(emitPower: number, worldMatrix: Matrix, directionToUpdate: Vector3, particle: Particle): void;
+        startPositionFunction(worldMatrix: Matrix, positionToUpdate: Vector3, particle: Particle): void;
+    }
+
+    export class BoxPartcileEmitter implements IParticleEmitterType {
```
Comment (deltakosh) [inline src/Particles/babylon.boxParticleEmitter.ts]: Small typo

File: src/Particles/babylon.coneParticleEmitter.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+module BABYLON {
+    export class ConePartcileEmitter implements IParticleEmitterType {
```
Comment (deltakosh) [inline src/Particles/babylon.coneParticleEmitter.ts]: Typo:)

File: src/Particles/babylon.particleSystem.ts [hunk 5]
```diff
@@ -623,6 +598,37 @@
             return result;
         }
 
+        public createSphereEmitter(redius = 1) {
```
Comment (deltakosh) [inline src/Particles/babylon.particleSystem.ts]: Radius:)

File: src/Particles/babylon.particleSystem.ts [hunk 6]
```diff
@@ -623,6 +598,37 @@
...
 
+        public createSphereEmitter(redius = 1) {
+            this.particleEmitterType = new SpherePartcileEmitter(redius);
+        }
+
+        public createDirectedSphereEmitter(redius = 1, direction1 = new Vector3(0, 1.0, 0), direction2 = new Vector3(0, 1.0, 0)) {
+            this.particleEmitterType = new SphereDirectedPartcileEmitter(redius, direction1, direction2);
+        }
+
+        public createConeEmitter(redius = 1, angle = Math.PI / 4) {
```
Comment (deltakosh) [inline src/Particles/babylon.particleSystem.ts]: Same here: radius

File: src/Particles/babylon.sphereParticleEmitter.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
+module BABYLON {
+    export class SpherePartcileEmitter implements IParticleEmitterType {
```
Comment (deltakosh) [inline src/Particles/babylon.sphereParticleEmitter.ts]: Typo

**Comment by IbraheemOsama** [comment]
Will create a new PR was merging some changes and reverted by mistake to old commit :(

**Pull request closed without merging**